src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,33 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
-        $seen = [];
+        if (\count($envelope->all(OverrideSendersStamp::class)) > 0) {
+            $overrideSendersStamp = $envelope->last(OverrideSendersStamp::class);
```

Comment:
Would be great to have tests for this section section

Reply:
Once we finalize the approach, I will add the tests